lib/catalog.js

```diff
@@ -22,13 +22,12 @@ const reload = (conn, provider) => {
   }).then(httpBody);
 };
 
-const addCredential = (conn, user, label) => {
+const addCredential = (conn, credentials, label) => {
   const headers = conn.headers();
   headers.set('Content-Type', 'application/json');
 
   const body = {
-    username: user.username,
-    password: user.password,
+    ...credentials,
```

Comment:
Works for me, but should we allow the user to set whatever on this? (I don't know if the way we do it normally is from inertia, or security, or etc.).